src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/handlers/presubmit/RequestEvidenceBundleNotificationHandler.java

```diff
@@ -52,58 +53,36 @@ public boolean canHandle(PreSubmitCallbackStage callbackStage, Callback<AsylumCa
-            String documentReference = homeOfficeCaseStatus.get().getApplicationStatus().getDocumentReference();
-            if (!isEmpty(documentReference)) {
-                log.info("Using document reference {} from application status instead of home{}",
-                    documentReference, homeOfficeReferenceNumber);
-                homeOfficeReferenceNumber = documentReference;
-            }
-        }
+        final String homeOfficeReferenceNumber = getHomeOfficeReference(asylumCase);
 
         final String caseId = asylumCase.read(APPEAL_REFERENCE_NUMBER, String.class)
             .orElseThrow(() -> new IllegalStateException("Case ID for the appeal is not present"));
 
-        final String correlationId = HomeOfficeRequestUuidGenerator.generateUuid();
-
         final RequestEvidenceBundleInstructMessage bundleInstructMessage =
-            buildRequest(caseId, correlationId, homeOfficeReferenceNumber, asylumCase);
-
-        final String notificationStatus = homeOfficeInstructService.sendNotification(bundleInstructMessage);
-        asylumCase.write(AsylumCaseDefinition.HOME_OFFICE_INSTRUCT_STATUS, notificationStatus);
-
-        return new PreSubmitCallbackResponse<>(asylumCase);
-    }
+            requestEvidenceBundleInstructMessage()
+                .withConsumerReference(getConsumerReference(caseId))
+                .withHoReference(homeOfficeReferenceNumber)
+                .withMessageHeader(getMessageHeader())
+                .withMessageType(REQUEST_EVIDENCE_BUNDLE.name())
+                .withChallenge(buildHomeOfficeChallenge(asylumCase))
+                .withNote(getDirectionContent(asylumCase, DirectionTag.RESPONDENT_EVIDENCE))
+                .withDeadlineDate(getDirectionDeadline(asylumCase, DirectionTag.RESPONDENT_EVIDENCE))
+                .build();
 
-    private RequestEvidenceBundleInstructMessage buildRequest(
-        String caseId, String correlationId, String homeOfficeReferenceNumber, AsylumCase asylumCase
-    ) {
-
-        RequestEvidenceBundleInstructMessageBuilder messageBuilder = requestEvidenceBundleInstructMessage();
-
-        homeOfficeInstructService.buildCoreAttributes(
-            messageBuilder, REQUEST_EVIDENCE_BUNDLE, homeOfficeReferenceNumber, caseId, correlationId
-        );
+        log.info("Finished constructing {} notification request for caseId: {}, HomeOffice reference: {}",
+            REQUEST_EVIDENCE_BUNDLE.toString(), caseId, homeOfficeReferenceNumber);
 
-        homeOfficeInstructService.extractDirectionAttributes(
-            messageBuilder, asylumCase, DirectionTag.RESPONDENT_EVIDENCE);
+        final String notificationStatus = homeOfficeInstructService.sendNotification(bundleInstructMessage);
 
-        HomeOfficeChallenge challenge = homeOfficeInstructService.buildHomeOfficeChallenge(asylumCase);
-        messageBuilder.withChallenge(challenge);
+        asylumCase.write(AsylumCaseDefinition.HOME_OFFICE_INSTRUCT_STATUS, notificationStatus);
```

Comment:
status should be set for Request Evidence specifically

Reply:
See the above comment.